packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1286,42 +1360,193 @@ export class EthModule {
-      //  refactored away
-      // TODO: This will log the error, but the RPC method won't be red
-      this._logError(error);
+      throw sentTxError;
+    }
+  }
+
+  private async _logBlock(
+    result: MineBlockResult,
+    sentTx: Transaction,
+    manyTransactionsLogged: boolean
+  ) {
+    const { block, blockResult, traces } = result;
+
+    if (manyTransactionsLogged) {
+      this._logBlockNumber(result.block);
+    }
+
+    for (let i = 0; i < block.transactions.length; i++) {
+      const tx = block.transactions[i];
+      const txGasUsed = bufferToInt(blockResult.receipts[i].gasUsed);
+      const txTrace = traces[i];
+
+      const boldTxHash =
+        manyTransactionsLogged && tx.hash().equals(sentTx.hash());
+
+      await this._logBlockTransaction(
+        tx,
+        txGasUsed,
+        txTrace,
+        block,
+        manyTransactionsLogged,
+        boldTxHash
+      );
+
+      this._printEmptyLineBetweenTransactions(i, block.transactions.length);
+    }
+  }
+
+  private async _logBlockTransaction(
+    tx: Transaction,
+    txGasUsed: number,
+    txTrace: GatherTracesResult,
+    block: Block,
+    manyTransactionsLogged: boolean,
+    boldTxHash: boolean
+  ) {
+    this._logger.setIndentEnabled(manyTransactionsLogged);
+
+    await this._logTransactionTrace(
+      tx,
+      txTrace.trace,
+      block,
+      txGasUsed,
+      manyTransactionsLogged,
+      boldTxHash
+    );
+
+    if (txTrace.trace !== undefined) {
+      await this._runHardhatNetworkMessageTraceHooks(txTrace.trace, false);
+    }
+
+    this._logConsoleLogMessages(txTrace.consoleLogMessages);
+
+    if (txTrace.error !== undefined) {
+      this._logError(txTrace.error);
+    }
+
+    this._logger.setIndentEnabled(false);
+  }
+
+  private async _logCurrentlySentTransaction(
+    tx: Transaction,
+    txReceipt: TxReceipt,
+    txTrace: GatherTracesResult,
+    block: Block
+  ) {
+    const txGasUsed = bufferToInt(txReceipt.gasUsed);
+    this._logger.log(chalk.bold.yellow("\nCurrently sent transaction:"));
+    await this._logBlockTransaction(tx, txGasUsed, txTrace, block, false, true);
+  }
+
+  private _getTransactionIndex(
+    tx: Transaction,
+    result: MineBlockResult
+  ): number {
+    const transactions = result.block.transactions;
+    for (let i = 0; i < transactions.length; i++) {
+      const blockTx = transactions[i];
+      if (blockTx.hash().equals(tx.hash())) {
+        return i;
+      }
     }
 
-    return bufferToRpcData(tx.hash(true));
+    throw new Error(
+      "The sent transaction not found in sendTransaction result, this should never happen"
+    );
+  }
+
+  private _logMultipleTransactionsWarning() {
+    this._logger.log(
+      "There were other pending transactions mined in the same block:\n"
+    );
+  }
+
+  private _logMultipleBlocksWarning() {
+    this._logger.log(
+      chalk.bold.yellow(
+        "There were other pending transactions. More than one block had to be mined:"
```

Comment:
Is this message correct?

If I disable automining, send two transactions, enable it again, and send a third one, all of them are mined in the same block, so there wasn't "more than one block" mine.